.github/workflows/lint_and_test.yml

```diff
@@ -0,0 +1,42 @@
+name: Lint and Test
+
+on: [push, pull_request]
+
+jobs:
+  fmt:
+    name: fmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - uses: actions/checkout@v4
+      - run: cargo fmt -- --check
+
+  clippy:
+    name: clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+      - uses: actions/checkout@v4
+      - run: cargo clippy --all --all-targets --all-features
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - run: cargo build --all-features
+      - run: cargo test --verbose --all --all-features
```

Comment:
It might be nice to run tests and clippy without features to check that the feature gate are correct